docs/requests.md

```diff
@@ -116,6 +116,45 @@ In some cases such as long-polling, or streaming responses you might need to
 determine if the client has dropped the connection. You can determine this
 state with `disconnected = await request.is_disconnected()`.
 
+#### Request Files
+
+Request files are normally sent as multipart form data (`multipart/form-data`).
+
+When you call `await request.form()` you receive a `starlette.datastructures.FormData` which is an immutable
+multidict, containing both file uploads and text input.
+
+When one of the fields is a file upload, you will get a `starlette.datastructures.UploadFile`.
+
+`UploadFile` has the following attributes:
+
+* `filename`: A `str` with the original file name that was uploaded (e.g. `myimage.jpg`).
+* `content_type`: A `str` with the content type (MIME type / media type) (e.g. `image/jpeg`).
+* `file`: A <a href="https://docs.python.org/3/library/tempfile.html#tempfile.SpooledTemporaryFile" target="_blank">`SpooledTemporaryFile`</a> (a <a href="https://docs.python.org/3/glossary.html#term-file-like-object" target="_blank">file-like</a> object). This is the actual Python file that you can pass directly to other functions or libraries that expect a "file-like" object.
+
+
+`UploadFile` has the following `async` methods. They all call the corresponding file methods underneath (using the internal `SpooledTemporaryFile`) in a threadpool.
+
+* `write(data)`: Writes `data` (`str` or `bytes`) to the file.
+* `read(size)`: Reads `size` (`int`) bytes/characters of the file.
+* `seek(offset)`: Goes to the byte position `offset` (`int`) in the file.
+    * E.g., `await myfile.seek(0)` would go to the start of the file.
+    * This is especially useful if you run `await myfile.read()` once and then need to read the contents again.
+* `close()`: Closes the file.
+
+As all these methods are `async` methods, you need to "await" them.
+
+For example, inside of an `async` route you can get the contents with:
+
+```Python
+contents = await myfile.read()
+```
+
+If you are inside of a normal `def` route, you can access the `UploadFile.file` directly, for example:
+
+```Python
+contents = myfile.file.read()
+```
```

Comment:
Let's not bother with this section.

Reply:
Sure! Removed.

Reply:
I think we can keep this example really short...

```python
form = await request.form()
filename = form["upload_file"].filename
contents = await form["upload_file"].read()
```

Don't really need to include a complete endpoint - just a snippet similar to the above would be fine.

Thanks so much for this!

Reply:
Get it.

Done! :heavy_check_mark:

> Thanks so much for this!

That's the least I can do :blush: . Thanks for Starlette! :rocket: